src/lib/next-auth.ts

```diff
@@ -37,32 +39,57 @@ export const options: NextAuthOptions = {
   pages: {
     error: '/auth/error', // Error code passed in query string as ?error=
   },
-  session: {
-    // Choose how you want to save the user session.
-    // The default is `"jwt"`, an encrypted JWT (JWE) stored in the session cookie.
-    // If you use an `adapter` however, we default it to `"database"` instead.
-    // You can still force a JWT session by explicitly defining `"jwt"`.
-    // When using `"database"`, the session cookie will only contain a `sessionToken` value,
-    // which is used to look up the session in the database.
-    strategy: 'database',
-
-    // Seconds - How long until an idle session expires and is no longer valid.
-    maxAge: 30 * 60, // 30 minutes
-
-    // Seconds - Throttle how frequently to write to database to extend a session.
-    // Use it to limit write operations. Set to 0 to always update the database.
-    // Note: This option is ignored if using JSON Web Tokens
-    updateAge: 15 * 60, // 15 minutes
-  },
   callbacks: {
-    signIn: async ({ user, account, profile }) => {
-      return true
+    async redirect({ url, baseUrl }) {
+      return baseUrl
     },
-    session: async ({ session, user, token }) => {
+
+    async session({ session, user }) {
+      if (!session || !user) return session
+
       session.user.id = user.id
+
+      const userAccounts = await prisma.account.findMany({
+        where: { userId: user.id },
+      })
+      const storedUserGitHub =
+        userAccounts.length === 2 &&
+        userAccounts.filter((account) => account.provider === 'github')
```

Comment:
instead of `.filter` we can use `.some` to check for a truthy value. This would also bypass the length check